src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -241,8 +252,45 @@ void IExtensionConfigProvider.Initialize(ExtensionConfigContext context)
             this.taskHubWorker = new TaskHubWorker(this.defaultDurabilityProvider, this, this);
             this.taskHubWorker.AddOrchestrationDispatcherMiddleware(this.EntityMiddleware);
             this.taskHubWorker.AddOrchestrationDispatcherMiddleware(this.OrchestrationMiddleware);
+
+#if !FUNCTIONS_V1
+            // The RPC server needs to be started sometime before any functions can be triggered
+            // and this is the latest point in the pipeline available to us.
+            this.StartLocalRcpServer();
+#endif
+        }
+
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            this.HttpApiHandler?.Dispose();
+        }
+
+#if !FUNCTIONS_V1
+        private void StartLocalRcpServer()
+        {
+            bool? shouldEnable = this.Options.LocalRpcEndpointEnabled;
+            if (!shouldEnable.HasValue)
+            {
+                // Default behavior is to only start the local RPC server for non-.NET function languages.
+                // We'll enable it if it's non-.NET or if the FUNCTIONS_WORKER_RUNTIME value isn't present.
+                string functionsWorkerRuntime = this.nameResolver.Resolve("FUNCTIONS_WORKER_RUNTIME");
+                shouldEnable = !string.Equals(functionsWorkerRuntime, "dotnet", StringComparison.OrdinalIgnoreCase);
+            }
+
+            // This is only applicable to Functions 2.0 and above
+            if (shouldEnable == true)
+            {
+                this.HttpApiHandler.StartLocalHttpServerAsync().GetAwaiter().GetResult();

```

Comment:
From my experience, this is the best way to handle it. It ensures that if an exception is thrown, we get the actual exception we care about rather than an AggregateException, which is what `.Wait()` or `.Result` would throw if there was a failure.